include/graphit/midend/mir.h

```diff
@@ -907,7 +910,9 @@ namespace graphit {
 
         struct EdgeSetApplyExpr : public ApplyExpr {
             std::string from_func = "";
+            std::vector<std::string> fromFuncFunctorArgs;
```

Comment:
Instead of adding the args to each ApplyExpr, you should add a FuncExpr to each of these classes. That will encapsulate the args and the name of the function together. I thought FuncExpr was for this reason. I see there is no FuncExpr in MIR. I think if we are going with FuncExpr, we should use it all the way down.

Reply:
Yeah that makes sense. I guess we have to rewrite the edgeset generation then? Instead of having a helper functions that return string, we do something else. @yunmingzhang17 what do you think?

Reply:
Yeah, I think Ajay is right on this one. We should make a MIR FuncExpr. I don't think this should affect code generation too much? It is pretty similar to what you have now, except the fromFuncFunctorArgs will be encapsulated in a MIR FuncExpr.

Reply:
sounds good. will do